lib/cchain/provider/provider.go

```diff
@@ -72,6 +72,7 @@ type Provider struct {
 	backoffFunc func(context.Context) func()
 	chainNamer  func(xchain.ChainVersion) string
 	network     netconf.ID
+	cache       *attestationCache
```

Comment:
as discussed in the meet yesterday, what about putting this inside the `stream` package rather?

Reply:
I definitely have no strong feelings yet, so I'll move there.

Reply:
Interesting observation: since `Deps` is generic (supports `any E`), in order to wrap the generic `FetchBatch` with a internal cached version, my cache needs to work for `any E` too, yet I have a requirement for some simple interface that can give me the key of the value. Then we could implement this interface for Attestations to return the `AttestOffset` as key and do the same for whatever else is used with `Stream[E any]`.

This is a bigger change and I can definitely do it, but wanted to make you aware that the footprint of the change will be much larger.